platform/linux-generic/odp_comp.c

```diff
@@ -11,24 +11,457 @@
 #include <odp/api/event.h>
 #include <odp/api/packet.h>
 #include <odp/api/plat/strong_types.h>
+#include <odp_packet_internal.h>
 
 #include <odp_debug_internal.h>
+#include <odp_init_internal.h>
+
+#include "miniz.h"
+
+#define MAX_SESSIONS  16
+#define MEM_LEVEL   8
+
+/** Forward declaration of session structure */
+typedef struct odp_comp_generic_session odp_comp_generic_session_t;
+
+#define to_gen_session(s) ((odp_comp_generic_session_t *)(intptr_t)(s))
+
+/**
+ * Algorithm handler function prototype
+ */
+typedef
+int (*comp_func_t)(odp_packet_t pkt_in,
+		   odp_packet_t pkt_out,
+		   const odp_comp_packet_op_param_t *params,
+		   odp_comp_generic_session_t *session);
+
+/**
+ * Per session data structure
+ */
+struct odp_comp_generic_session {
+	struct odp_comp_generic_session *next;
+	odp_comp_session_param_t        params;
+	struct {
+		comp_func_t func;
+		mz_stream stream;
+		union {
+			tdefl_compressor comp;
+			inflate_state inflate;
+		} data;
+	} comp;
+};
+
+typedef struct odp_crypto_global_s {
+	odp_spinlock_t                lock;
+	odp_comp_generic_session_t *free;
+	odp_comp_generic_session_t  sessions[MAX_SESSIONS];
+} odp_crypto_global_t;
```

Comment:
comp_global_t ... not crypto

Reply:
I should stop doing C&P while having a cold.